### PR TITLE
[NPUW] Extend ConvToMatmul pattern

### PR DESCRIPTION
**Pull request by smirnov-alexey**
Support case where Const has no Convert after it

### CODE REVIEW

File: src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp [hunk 1]
```diff
@@ -1501,6 +1501,7 @@ ConvToMatmul::ConvToMatmul(Context::Ref ctx) {
         auto matched_node_transpose_in = node_to_output.at(transpose_in).get_node_shared_ptr();
         auto matched_node_transpose_out = node_to_output.at(transpose_out).get_node_shared_ptr();
         auto matched_node_multiply = node_to_output.at(multiply).get_node_shared_ptr();
+        auto matched_node_param2_out = uat::_(node_to_output).at_or_at(convert2, multiply);
```
Comment (AsyaPronina) [inline src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp:1504]: it might be useful to use `const &` here to avoid Coverity complains

Comment (smirnov-alexey) [reply src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp]: Done

File: src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp [hunk 2]
```diff
@@ -1501,6 +1501,7 @@ ConvToMatmul::ConvToMatmul(Context::Ref ctx) {
         auto matched_node_transpose_in = node_to_output.at(transpose_in).get_node_shared_ptr();
         auto matched_node_transpose_out = node_to_output.at(transpose_out).get_node_shared_ptr();
         auto matched_node_multiply = node_to_output.at(multiply).get_node_shared_ptr();
+        const auto& matched_node_param2_out = uat::_(node_to_output).at_or_at(convert2, multiply);
```
Comment (dmatveev) [inline src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp:1504]: I'd call it `cvt2_or_multiply` - as it is exactly what you're looking up for in the map.

File: src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp [hunk 3]
```diff
@@ -1536,10 +1537,10 @@ ConvToMatmul::ConvToMatmul(Context::Ref ctx) {
             auto new_reshape2 = std::make_shared<ov::op::v1::Reshape>(matched_node_param2, new_const2, false);
 
             // Connect to Reshape
-            if (ov::op::util::is_parameter(matched_node_param2)) {
+            if (matched_node_param2_out == matched_node_multiply) {
```
Comment (dmatveev) [inline src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp:1540]: ...then the logic here becomes clear

```suggestion
            if (cvt2_or_multiply == matched_node_multiply) {
```

File: src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp [hunk 4]
```diff
@@ -1536,10 +1537,10 @@ ConvToMatmul::ConvToMatmul(Context::Ref ctx) {
...
                 matched_node_multiply->input(1).replace_source_output(new_reshape2);
                 matched_node_multiply->validate_and_infer_types();
-            } else {  // constant -> convert -> multiply
+            } else {
```
Comment (dmatveev) [inline src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp:1543]: But some comments documenting what cases you're looking at still would help a lot

**Review by dmatveev — COMMENTED** [review]
One-lines get most of the attention

File: src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp (hunk 2 above)
Comment (smirnov-alexey) [reply src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp]: Done

File: src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp (hunk 3 above)
Comment (smirnov-alexey) [reply src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp]: Done

File: src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp (hunk 4 above)
Comment (smirnov-alexey) [reply src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp]: Done

**Review by dmatveev — APPROVED** [review]

**Pull request merged**